### ISSUE TITLE
Drop extension when saving user files

### ISSUE THREAD

**Issue by jonasraoni** [pkp/pkp-lib#7696]
**Describe the problem you would like to solve**
At this moment, we're saving files with their extensions (`.php`, `.htaccess` and other variants are renamed)... Given that servers might be able to execute other types of files, a safer approach, in case the user has a public `files_dir`, would be to get rid of the extensions.

**Describe the solution you'd like**
Remove the extension from the file name.
Considering that we already have everything that's needed in place (filename generator + script proxying the download), the impact seems to be small.

**Who is asking for this feature?**
Developer

**Comment by asmecher** [comment]
Even if a file isn't executable, an exposed `files_dir` will allow 3rd parties who can guess filenames to download submission files without PHP-land access checks. I'm in favour of having clear instructions (*never make your files directory public*) rather than ambiguity.

**Comment by jonasraoni** [comment]
Yep! My intention here was just to avoid something more catastrophic for the worst case (open `files_dir` in a server configured to execute other extensions).
